query-engine/connector-test-kit-rs/query-engine-tests/tests/new/invalid_input_error.rs

```diff
@@ -0,0 +1,81 @@
+use query_engine_tests::*;
+
+/*
+// TODO: uncomment this test when we figure out how to properly override environment variables
+// for a single test. For the moment, we're asserting the user-facing error message in TypeScript tests.
+
+#[test_suite(schema(generic), only(Postgres))]
+mod find_many {
+    #[connector_test]
+    async fn value_too_large_to_transmit(runner: Runner) -> TestResult<()> {
+        let n = 32767;
+
+        // [1,2,...,n]
+        let ids: Vec<u32> = (1..n + 1).collect();
+
+        let query = format!(
+            r#"
+            query {{
+              findManyTestModel(where: {{
+                id: {{
+                  in: {:?}
+                }}
+              }}) {{
+                id
+              }}
+            }}"#,
+            ids,
+        );
+
+        let old_query_batch_size = std::env::var("QUERY_BATCH_SIZE");
+        std::env::set_var("QUERY_BATCH_SIZE", n.to_string());
+        assert_error!(
+            runner,
+            query,
+            2034,
+            "Assertion violation on the database: `value too large to transmit`"
+        );
+        std::env::set_var("QUERY_BATCH_SIZE", old_query_batch_size.unwrap_or_default());
+
+        Ok(())
+    }
+}
+*/
+
+#[test_suite(schema(generic), only(Postgres))]
+mod raw_params {
+    #[connector_test]
+    async fn value_too_large_to_transmit(runner: Runner) -> TestResult<()> {
+        let n = 32768;
+
+        // [1,2,...,n]
+        let ids: Vec<u32> = (1..n + 1).collect();
+
+        // "$1,$2,...,$n"
+        let params: String = ids
+            .iter()
+            .map(|id| format!("${}", id))
+            .collect::<Vec<String>>()
+            .join(",");
+
+        let mutation = format!(
+            r#"
+            mutation {{
+              queryRaw(
+                query: "SELECT * FROM \"TestModel\" WHERE id IN ({})",
+                parameters: "{:?}"
+              )
+            }}"#,
+            params, ids,
+        );
+
+        assert_error!(
+            runner,
+            mutation,
+            2034,
+            "Assertion violation on the database: `value too large to transmit`"
```

Comment:
Is this an error that actually helps our users to understand that they need to change something?

Reply:
Or, we send an error that hey, you sent us more than i16::MAX records. Please change your IN statement. I think that's more clear for the user...